### ISSUE TITLE
Add timeout support for aws_sagemaker_project

### ISSUE THREAD

**Issue by george-chenvibes** [hashicorp/terraform-provider-aws#25637]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Deploying a Sagemaker Project takes a few minutes. But the creation of a project stops after 2m saying:
`error waiting for SageMaker Project (data-ml-code-commit25) to be created: timeout while waiting for state to become 'CreateCompleted' (last state: 'CreateInProgress', timeout: 2m0s)`
### New or Affected Resource(s)

* aws_sagemaker_project

### Potential Terraform Configuration

```hcl
resource "aws_sagemaker_project" "sagemaker_project" {
  name                 = "my-project"
  service_catalog_provisioning_details {
    product_id = aws_servicecatalog_product.example.id
  }
  timeouts {
    create = "30m"
  }
}
```

### References

* #0000

**Comment by bschaatsbergen** [comment]
@george-chenvibes instead of introducing a timeout I can increase the timeout from 2 to 15 minutes. Would that help solve your problem?

**Comment by george-chenvibes** [comment]
@bschaatsbergen That will help. I'll see if it works after you increase the timeout.

**Comment by bschaatsbergen** [comment]
Hi @george-chenvibes, this has been merged and will be part of the next (v4.22.0) release.

**Comment by bschaatsbergen** [comment]
> Hi @bschaatsbergen, do you know when that release will occur?

We just released, I reckon somewhere next week.
